app/tests/unit/forms/auth/authService.spec.js

```diff
@@ -0,0 +1,81 @@
+const service = require('../../../../src/forms/auth/service');
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('parseToken', () => {
+
+  it('returns a default object when an exception happens', () => {
+    const result = service.parseToken(undefined);
+    expect(result).toEqual({
+      keycloakId: undefined,
+      username: 'public',
+      firstName: undefined,
+      lastName: undefined,
+      fullName: 'public',
+      email: undefined,
+      idp: 'public',
+      public: true
+    });
+  });
+});
+
+describe('formAccessToForm', () => {
+  it('transforms the object', () => {
+    const result = service.formAccessToForm({
+      formId: 1,
+      formName: 2,
+      formDescription: 3,
+      labels: 4,
+      idps: 5,
+      active: 6,
+      formVersionId: 7,
+      version: 8,
+      published: 9,
+      versionUpdatedAt: 10,
+      roles: 11,
+      permissions: 12
+    });
+    expect(result).toEqual({
+      formId: 1,
+      formName: 2,
+      formDescription: 3,
+      labels: 4,
+      idps: 5,
+      active: 6,
+      formVersionId: 7,
+      version: 8,
+      published: 9,
+      versionUpdatedAt: 10,
+      roles: 11,
+      permissions: 12
+    });
```

Comment:
It looks like those two objects are identical, but declared explicitly twice. You could do something like the following to make it more concise:
```js
const form = {
  formId: 1,
  formName: 2,
  ...
  permissions: 12
};
const result = service.formAccessToForm(form);
expect(result).toEqual(form);
```

Reply:
That'll be a reference equality though right? And the point of the test is to see that the contents of the 2 objects are the same and not that the reference is just being returned. (or does `toEqual` de-reference?)

Reply:
NM, answered my own question. So I think you're right it can be shortedned
https://stackoverflow.com/questions/45195025/what-is-the-difference-between-tobe-and-toequal-in-jest